ui/desktop/app/templates/scopes/scope/projects/sessions/session.hbs

```diff
@@ -15,13 +15,15 @@
   </page.header>
 
   <page.actions>
-    <Hds::Button
-      data-test-session-detail-cancel-button
-      @text={{t 'resources.session.actions.end'}}
-      @icon='x'
-      @color='secondary'
-      {{on 'click' (route-action 'cancelSession' @model)}}
-    />
+    {{#if (can 'cancel session' @model)}}
```

Comment:
None blocking comment: same comment as above about disabling the button rather than hidding it.